server/db/models/player.js

```diff
@@ -148,204 +89,183 @@ playerSchema.methods.setCardinal = function(row, col) {
-  return this.model('Player')
-  .find({game: this.game})
-  .where({_id :{$ne: this._id}})
-  .bind(this);
-};
-
 playerSchema.methods.boardMove = function (bearing) {
   // call when self.bearing is not relevant
   var newCol = this.position[0];
   var newRow = this.position[1];
   newCol += bearing[0];
   newRow += bearing[1];
-  return
-  this.set('position', [newCol, newRow]);
+  // check if move is possible
+  if (newCol < 0 || newRow < 0 || newCol > 11 || newRow > 15) return this.loseLife();
+  else {
+    this.set('position', [newCol, newRow]);
+  }
 };
 
-playerSchema.methods.pushPlayer = function (){
-  return this.getOpponents().bind(this)
-  .then(function(opponents){
-    return Promise.map(opponents, function(o){
-      // if
-    })
-  })
+
+playerSchema.methods.loseLife = function() {
+  this.livesRemaining--;
+  if (this.livesRemaining === 0) return this.killPlayer();
+  else {
+    this.set('position', this.dock);
+  }
 };
 
-playerSchema.methods.checkMove = function(bearing) {
-  var key = 'edge' + bearing[2];
-  return this.findMyTile()
-  .then(function(tile) {
-    if (tile[key] === null) return true;
-    else return false
-  })
+
+playerSchema.methods.killPlayer = function() {
+  this.position = null;
+};
+
+
+function findOppositeBearing(bearing) {
+  if (bearing == [-1,0,'N']) return [1,0,'S'];
+  if (bearing == [0,1,'E']) return [0,-1,'W'];
+  if (bearing == [1,0,'S']) return [-1,0,'N'];
+  if (bearing == [0,-1,'W']) return [0,1,'E'];
+}
+
+
+playerSchema.methods.checkMove = function() {
+  var currentKey = 'edge' + this.bearing[2];
+  var oppositeBearing = findOppositeBearing(this.bearing)
+  var nextKey = 'edge' + oppositeBearing[2]
+
+  var currentPosition = this.position;
+  var nextPosition = [this.position[0] + this.bearing[0], this.position[1] + this.bearing[1]]
+
+  var currentTile = this.game.getTileAt(currentPosition)
+  var nextTile = this.game.getTileAt(nextPosition)
+
+  if (currentTile[currentKey] === null && nextTile[nextKey] === null) return true;
+  else return false;
+}
+
+
+//do we need to add anything else to this function?
+//potentially use to deal with pushing adjacent players
+playerSchema.methods.oneCardMove = function(bearing) {
+  var newRow, newCol, opponent;
+
+  var checkMove = this.checkMove()
+
+  if (checkMove === true) {
+    var adjacentOpponent = this.game.getPlayerAt([this.position[0]+this.bearing[0], this.position[1]+this.bearing[1]])
+    if (adjacentOpponent) return adjacentOpponent.boardMove(this.bearing)
+    else return 
+  }
 }
 
+
 playerSchema.methods.cardMove = function (magnitude) {
   var newCol = this.position[1];
   var newRow = this.position[0];
 
-  // check that move is permitted
-  return this.checkMove()
-  .then(function(result){
-    if (result === true){
+  var checkMove = this.checkMove()
 
+  if (checkMove === true) {
+    while (magnitude > 0) {
+      newCol += this.bearing[1];
+      if (newCol < 0 || newCol > 11) return this.loseLife();
+
```

Comment:
It seems that checkPit could check for an actual pit and a player falling off the edge. That way it could be one function, I think.